_posts/2022-01-13-spice-up-your-javascript-with-curry.md

```diff
@@ -154,8 +154,19 @@ const salary = (employee) => employee.salary;
 // We need a function that sorts given list in ascending order
 const sortAscending = (list) => list.sort((a, b) => a - b);
 
-// We need a function that returns the item in the center of list
-const median = (list) => list[Math.floor(list.length / 2)];
+// We need a function that returns the middle item
+const medianOdd = (list) => list[Math.floor(list.length / 2)];
+
+// And a function that returns average of summed middle items even numbered lists
+const medianEven = (list) => {
+  const middleIndex = Math.floor(list.length / 2);
+  return (list[middleIndex] + list[middleIndex + 1]) / 2;
+};
+
+// We need a function that returns the item in the center of list or average of
+// 2 middle items in lists that contain even number of items
+const median = (list) =>
+  list.length % 2 === 0 ? medianEven(list) : medianOdd(list);
```

Comment:
Nice correction :). I thought to mention about this initially but thought that it is a too tinyt detail that doesn't affect to the actual content of this blog post.

Reply:
Just had to correct it, due to this https://xkcd.com/386/ 😉